### PR TITLE
Handle Duplicates

### PR DESCRIPTION
**Pull request by AlagappanRa**
- Duplicates are defined as "entries with the same company name and the same role"
- Add and Edit Command edited
- Associated Test Cases (New - added)
- Old Test Cases Modified

### CODE REVIEW

File: src/main/java/seedu/address/model/company/Company.java [hunk 1]
```diff
@@ -130,4 +135,13 @@ public String toString() {
...
 
+    /**
+     * Returns a string representation of the company in the form of a list of attributes.
+     * Used to display the entire string representation in one line
+     */
+    public String styledOutput() {
+        return "Name=" + name + "," + " Role=" + role + "," + " Status=" + status + ","
+                + " Deadline=" + deadline + "," + " Recruiter Name=" + recruiterName + ","
+                + " Phone=" + phone + "," + " Email=" + email + "," + " Priority=" + priority;
+    }
```
Comment (papataco14) [inline src/main/java/seedu/address/model/company/Company.java:146]: Remove fields that are not necessary to be displayed to user, when they see a duplicate company error message.

**Review by papataco14 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -78,6 +78,14 @@ public boolean hasCompany(Company company) {
...
     }
 
+    /**
+     * Returns true if a company with the same identity as {@code company} exists in the address book.
+     */
+    public Company getDuplicateCompany(Company company) {
+        requireNonNull(company);
+        return companies.getDuplicateCompany(company);
+    }
+
```
[the remark this answers is not in the document]
Comment (AlagappanRa) [reply src/main/java/seedu/address/model/AddressBook.java:88]: Thanks.

File: src/main/java/seedu/address/model/company/Company.java [hunk 3]
```diff
@@ -72,16 +72,22 @@ public Priority getPriority() {
...
      */
     public boolean isSameCompany(Company otherCompany) {
-        if (otherCompany == this) {
-            return true;
+        if (otherCompany == null) {
+            return false;
         }
+        //Defensive Programming
+        Name otherName = otherCompany.getName();
+        Role otherRole = otherCompany.getRole();
```
[the remark this answers is not in the document]
Comment (AlagappanRa) [reply src/main/java/seedu/address/model/company/Company.java:84]: ;)

File: src/main/java/seedu/address/model/company/Company.java [hunk 4]
```diff
@@ -72,16 +72,22 @@ public Priority getPriority() {
...
 
         return otherCompany != null
```
[the remark this answers is not in the document]
Comment (AlagappanRa) [reply src/main/java/seedu/address/model/company/Company.java:86]: ;)

File: src/main/java/seedu/address/commons/util/FileUtil.java [hunk 5]
```diff
@@ -18,9 +18,9 @@ public static boolean isFileExists(Path file) {
     }
 
     /**
-     * Returns true if {@code path} can be converted into a {@code Path} via {@link Paths#get(String)},
-     * otherwise returns false.
-     * @param path A string representing the file path. Cannot be null.
+     * Returns true if the given string is a valid path.
+     * @param path
+     * @return true if the given string is a valid path.
```
[the remark this answers is not in the document]
Comment (AlagappanRa) [reply src/main/java/seedu/address/commons/util/FileUtil.java:23]: No.

**Review by papataco14 — APPROVED** [review]

**Pull request merged**